client/main.go

```diff
@@ -1,49 +1,35 @@
 package main
 
 import (
-	"log"
-	"net/http"
 	"os"
 
-	"github.com/99designs/gqlgen/graphql/handler"
-	"github.com/99designs/gqlgen/graphql/playground"
-	"github.com/Shelex/grpc-go-demo/client/graph"
-	"github.com/Shelex/grpc-go-demo/client/graph/generated"
-	"github.com/Shelex/grpc-go-demo/proto"
-	"google.golang.org/grpc"
-	"google.golang.org/grpc/credentials"
+	client "github.com/Shelex/grpc-go-demo/client/service"
+	"github.com/Shelex/grpc-go-demo/config"
+	"github.com/Shelex/grpc-go-demo/logger"
+	log "github.com/sirupsen/logrus"
 )
 
-const defaultPort = "8080"
-const serverPort = "9000"
-
 func main() {
-	port := os.Getenv("PORT")
-	if port == "" {
-		port = defaultPort
-	}
-
-	creds, err := credentials.NewClientTLSFromFile("cert.pem", "")
-	if err != nil {
-		log.Fatal(err)
-	}
-	opts := []grpc.DialOption{grpc.WithTransportCredentials(creds)}
-	conn, err := grpc.Dial("localhost:"+serverPort, opts...)
-	if err != nil {
-		log.Fatal(err)
-	}
 	defer func() {
-		if err := conn.Close(); err != nil {
-			log.Fatalf("failed to close connection: %s", err)
+		if r := recover(); r != nil {
+			log.Fatal(r)
 		}
 	}()
-	client := proto.NewEmployeeServiceClient(conn)
 
-	srv := handler.NewDefaultServer(generated.NewExecutableSchema(generated.Config{Resolvers: graph.NewResolver(client)}))
+	cfg := config.GetEnv()
 
-	http.Handle("/", playground.Handler("GraphQL playground", "/query"))
-	http.Handle("/query", srv)
+	logger.Init(cfg)
+	log.Info("starting client...")
 
-	log.Printf("connect to http://localhost:%s/ for GraphQL playground", port)
-	log.Fatal(http.ListenAndServe(":"+port, nil))
+	conn, err := client.ConnectGRPCService(cfg)
+	if err != nil {
+		log.Errorf("failed to create connection to server: %s", err)
+		os.Exit(1)
```

Comment:
A pattern I like is to use `os.Exit()` exactly once in the program in `main()`. You can do this by pushing all the code in main into another function that returns `error`.

Reply:
agree, will move to separate function.